dev-docs/security-overview.md

```diff
@@ -0,0 +1,74 @@
+
+The latter causes the first node to download, verify, and run these three services. The services' containers are hosted at https://ghcr.io/edgelesssys.
+
+After this, the Constellation cluster is operational on the first node.
+
+## Cluster growth
+
+Additional nodes can now join the cluster - either as control-plane nodes or as worker nodes. For both, the process for joining the cluster is identical. First, the Bootstrapper running on a *new node* contacts the JoinService of the existing cluster. The JoinService verifies the remote-attestation statement of the new node using the runtime measurements stored in etcd. On success, an aTLS connection between the two is created, which is used mainly for the following (see the the [interface definition](https://github.com/edgelesssys/constellation/blob/main/joinservice/joinproto/join.proto) for a comprehensive list of exchanged data):
+1. The new node sends a certificate signing request (CSR) to the JoinService.
+1. The JoinService issues a corresponding certificate and sends it to the new node. The JoinService uses the signing key of the Kubernetes cluster for this, which is [generated by kubeadm](https://kubernetes.io/docs/setup/best-practices/certificates/).
+1. The JoinService sends a [kubeadm token](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-token/) to the new node.
+1. The JoinService sends the hashes of the to-be-installed Kubernetes binaries to the new node.
+1. The JoinService sends encryption key for the new node's local storage. This key is generated and managed by the cluster's KeyService.
+1. The JoinService sends the certificate of the cluster's control plane to the new node.
+
+After this, the aTLS connection is closed and the node is marked as "initialized". 
+
+The Bootstrapper on the new node uses the kubeadm token to download the configuration of the cluster from the Kubernetes API server. Subsequently, it downloads, verifies, and runs the given Kubernetes binaries accordingly. The kubeadm token is never used after this. 
+
+The kubelet on the new node uses its own certificate and the certificate of the cluster's control plane (which the new node both received from the JoinService) to establish an mTLS connection with the cluster's Kubernetes API server. Once connected, the new node registers itself as control-plane node or worker node of the cluster. This process uses the standard Kubernetes mechanisms for adding nodes. 
+
+In Constellation, a virtual private network (VPN) exists between all nodes of a cluster. To join this VPN, the new node generates WireGuard credentials for itself and writes the public part to etcd via the mTLS connection with the Kubernetes API server. Whenever nodes within a Constellation cluster are talking to each other for the first time, they check etcd for the other node's public WireGuard credentials.

```

Comment:
> Whenever nodes within a Constellation cluster are talking to each other for the first time, they check etcd for the other node's public WireGuard credentials.

There is no synchronous mechanism to look up the public key before a packet is send out. The Cilium agent on the node watches for changes and eventually sets the correct public key and IPs on the WireGuard interface of the node. 
What we additionally do is that we forbid that unencrypted traffic leaves the node when it's source and destination reside in the same node subnet.